### DISCUSSION TITLE
Regarding resolver function for interface

### DISCUSSION THREAD

**Discussion by huanyu-li** [mirumee/ariadne#605] [Q&A] [answered]
Hi,

I just have a question regarding resolving interface when I read following docs (https://ariadnegraphql.org/docs/interfaces).
For following functions, as I understand Client, Order and Product correspond to the schema, but how are they imported here?

def resolve_search_result_type(obj, *_):
    if isinstance(obj, Client):
        return "Client"
    if isinstance(obj, Order):
        return "Order"
    if isinstance(obj, Product):
        return "Product"
    return None

In addition, I find sth from the main webpage like 'from .models import Category, Promotion'. Should Client, Order, and Product follow the same way to be imported like 'from .models import Client, Order, Product'?
However, It does not work in my case? Could you please help me to understand what's the meaning of 'from .models import ...'?

Thank you in advance.

Best,
Huanyu Li

**Comment by rafalp** [comment, marked as answer]
Hi! Thanks for asking!

What is important, is returning `str` with name of GraphQL type in your schema. Example decides what type name to return using `isinstance` check against few classes that were made up for the example and imported from some made up place (in this case `models`). It's up to you to decide where your classes will be defined and where you will import them from.

**Reply by huanyu-li** [reply]
Hi,

Thanks for the answer.
If I remember correctly, those Client, Order are classes defined corresponding to types in GraphQL schema.
In my case, I don't define such classes corresponding to types in GraphQL schema and I load the schema from a schema file (my_schema = load_schema_from_path("schema.graphql")). I suppose I need to have another way to decide which object type resolved for the interface. Is it correct?

Best,
Huanyu

**Reply by rafalp** [reply]
> I suppose I need to have another way to decide which object type resolved for the interface. Is it correct?

Yes, that's correct.

Common solution to this is embedding extra dict key (if you are using dicts) that contains typename:

```
data["__typename"] = "Client"
```

This code can live in resolver returning interface/union types.

**Reply by huanyu-li** [reply]
I get the point. Thank you so much.
